networkx/algorithms/tests/test_swap.py

```diff
@@ -3,7 +3,8 @@
 import networkx as nx
 
 cycle = nx.cycle_graph(5, create_using=nx.DiGraph)
-tree = nx.random_tree(10, create_using=nx.DiGraph, seed=42)
+tree = nx.DiGraph()
+tree.add_edges_from(nx.random_labeled_tree(10, seed=42).edges)
```

Comment:
I see now where the discussion about directed trees is coming from. It looks like this solution orients the edges according to the edge order of the `random_labeled_tree`. So AIUI that would be a `polytree` (the edges can be directed in any orientation). AIUI an arboresence forces the directions to be away from the root node. Hence at least some of the reason for punting on providing uniformly chosen random directed trees. (sorry for the spam on this PR -- nothing to change here)